### PR TITLE
BRS-424: Passes Bookable prior to 7am

### PR DESCRIPTION
**Pull request by Christopher-walsh22**
Ticket: [BRS-424](https://github.com/orgs/bcgov/projects/49/views/1?pane=issue&itemId=70969356)

- Reference system time from initial facility request to determine if the booking window is valid
- When outside booking window warning will be given to refresh page at appropriate time
- 
image

-If the time is within booking window user will be able to proceed to next window

### CODE REVIEW

**Review by davidclaveau — APPROVED** [review]

File: src/app/registration/facility-select/facility-select.component.spec.ts [hunk 1]
```diff
@@ -89,7 +90,7 @@ describe('FacilitySelectComponent', () => {
               '6': true,
             },
             bookableHolidays: [],
-            bookingDaysRichText: ''
+            bookingDaysRichText: '',
```
Comment (marklise) [inline src/app/registration/facility-select/facility-select.component.spec.ts:93]: No trailing commas

File: src/app/registration/facility-select/facility-select.component.spec.ts [hunk 2]
```diff
@@ -329,7 +330,7 @@ describe('FacilitySelectComponent', () => {
...
```
Comment (marklise) [inline src/app/registration/facility-select/facility-select.component.spec.ts:333]: Same here

File: src/app/registration/facility-select/facility-select.component.ts [hunk 3]
```diff
@@ -74,17 +74,23 @@ export class FacilitySelectComponent implements OnInit {
...
+  public allowedToBook: boolean;
+  public validBookingTime = true;
+  public formattedBookingTime: string | null;
 
   constructor(private fb: UntypedFormBuilder, private configService: ConfigService, private toastService: ToastService, private passService: PassService) { }
 
   ngOnInit(): void {
     if (this.configService) {
-      this.trailPassLimit = this.configService.config['TRAIL_PASS_LIMIT'];
+      this.trailPassLimit = this.configService.config['TRAIL_PASS_LIMIT'];
```
Comment (marklise) [inline src/app/registration/facility-select/facility-select.component.ts:87]: No extra space here

File: src/app/registration/facility-select/facility-select.component.ts [hunk 4]
```diff
@@ -112,20 +118,25 @@ export class FacilitySelectComponent implements OnInit {
...
+    this.formattedBookingTime = dateTime.toFormat("h:mm a 'PDT'");
     return bookingOpeningHour;
   }
 
-  get isOpeningHourPast(): boolean {
-    // check the current time in the America/Vancouver TZ (must do this step to acct for PST/PDT)
-    const currentHour = this.getPSTDateTime().get('hour');
-    return Boolean(parseInt(currentHour, 10) >= this.bookingOpeningHour);
+  get isOpeningHourPast(): boolean { 
+    if (this.systemTimePST != null){
```
Comment (marklise) [inline src/app/registration/facility-select/facility-select.component.ts:136]: NB: Spacing

`if (this.systemTimePST != null){` => `if (this.systemTimePST != null) {`

File: src/app/registration/facility-select/facility-select.component.ts [hunk 5]
```diff
@@ -261,6 +272,21 @@ export class FacilitySelectComponent implements OnInit {
     return DateTime.now().setZone('America/Vancouver');
   }
 
+  async getSystemTime(){
```
Comment (marklise) [inline src/app/registration/facility-select/facility-select.component.ts:275]: spacing

File: src/app/registration/facility-select/facility-select.component.ts [hunk 6]
```diff
@@ -261,6 +272,21 @@ export class FacilitySelectComponent implements OnInit {
...
+      const facilityWithCurrentTime = this.facilities.find(facility => facility.currentTime);
+      this.systemTime = facilityWithCurrentTime ? facilityWithCurrentTime.currentTime : null;
+      this.systemTimePST = this.convertUTCToPST(this.systemTime);
+      if (!this.isOpeningHourPast){
```
Comment (marklise) [inline src/app/registration/facility-select/facility-select.component.ts:279]: spacing

File: src/app/registration/facility-select/facility-select.component.ts [hunk 7]
```diff
@@ -261,6 +272,21 @@ export class FacilitySelectComponent implements OnInit {
...
+        this.validBookingTime = false;
+      } 
+  }
+
+  convertUTCToPST(utcTime){
```
Comment (marklise) [inline src/app/registration/facility-select/facility-select.component.ts:284]: spacing

**Review by marklise — APPROVED** [review]
👍🏻 /w nits.

File: src/app/registration/facility-select/facility-select.component.ts (hunk 7 above)
Comment (Christopher-walsh22) [reply src/app/registration/facility-select/facility-select.component.ts]: Resolved in [Commit43beab5](https://github.com/bcgov/parks-reso-public/pull/461/commits/43beab52a1c7d818910d777c2133aefa47ce60ff)

File: src/app/registration/facility-select/facility-select.component.ts (hunk 6 above)
Comment (Christopher-walsh22) [reply src/app/registration/facility-select/facility-select.component.ts]: Resolved in [Commit 43beab5](https://github.com/bcgov/parks-reso-public/pull/461/commits/43beab52a1c7d818910d777c2133aefa47ce60ff)

File: src/app/registration/facility-select/facility-select.component.ts (hunk 5 above)
Comment (Christopher-walsh22) [reply src/app/registration/facility-select/facility-select.component.ts]: Resolved in [Commit 43beab5](https://github.com/bcgov/parks-reso-public/pull/461/commits/43beab52a1c7d818910d777c2133aefa47ce60ff)

File: src/app/registration/facility-select/facility-select.component.ts (hunk 4 above)
Comment (Christopher-walsh22) [reply src/app/registration/facility-select/facility-select.component.ts]: Resolved in [Commit 43beab5](https://github.com/bcgov/parks-reso-public/pull/461/commits/43beab52a1c7d818910d777c2133aefa47ce60ff)

File: src/app/registration/facility-select/facility-select.component.ts (hunk 3 above)
Comment (Christopher-walsh22) [reply src/app/registration/facility-select/facility-select.component.ts]: Resolved in [Commit 43beab5](https://github.com/bcgov/parks-reso-public/pull/461/commits/43beab52a1c7d818910d777c2133aefa47ce60ff)

File: src/app/registration/facility-select/facility-select.component.spec.ts (hunk 2 above)
Comment (Christopher-walsh22) [reply src/app/registration/facility-select/facility-select.component.spec.ts]: Resolved in [Commit 43beab5](https://github.com/bcgov/parks-reso-public/pull/461/commits/43beab52a1c7d818910d777c2133aefa47ce60ff)

File: src/app/registration/facility-select/facility-select.component.spec.ts (hunk 1 above)
Comment (Christopher-walsh22) [reply src/app/registration/facility-select/facility-select.component.spec.ts]: Resolved in [Commit 43beab5](https://github.com/bcgov/parks-reso-public/pull/461/commits/43beab52a1c7d818910d777c2133aefa47ce60ff)

**Pull request merged**